scripts/react_native_pods.rb

```diff
@@ -61,6 +61,10 @@ def use_react_native! (
   app_path: '..',
   config_file_dir: '')
 
+  # Current target definition is provided by Cocoapods and it refers to the target
+  # that has invoked the `use_react_native!` function.
+  ReactNativePodsUtils.detect_use_frameworks(current_target_definition)
```

Comment:
do you know the sequence for cocoapods to read all podspecs and this `use_react_native!()` call? is it possible cocoapods processes podspec before this?

Reply:
This call is inside the `use_react_native!()` and it happens immediately. 

All the internal RN pods will benefit from this environment variable. If a user define a dependency before calling `use_react_native!()`, that podspec won't have the `ENV` variable set. However, we consider this a wrong setup. For a React Native app, you should first call `use_react_native!` and then add your other dependencies.

Does it make sense?

Reply:
in this case i would like to tell users that, for react-native projects, you should use `ENV['USE_FRAMEWORKS']` as single source of truth. so maybe we can change the detection for inconsistent setup for `current_target_definition.build_type`. e.g.,`if build_type.to_s == 'static framework' && ENV['USE_FRAMEWORKS'] == nil` we could show an error message that to enable use_frameworks!, please add by ENV['USE_FRAMEWORKS'].

after that even that podspec before use_react_native!() could use the ENV['USE_FRAMEWORKS'] to check.

Reply:
that's fair enough to me. your design also comes with an advantage when multiple targets in one Podfile, they could have different use_frameworks settings.

i test with the following Podfile and make sure the `use_react_native!` executed before podspec, which is great. i think the pr is good to go 🚀
```
target 'RNNext' do
  pod 'ExpoModulesCore', path: '../node_modules/expo-modules-core'

  config = use_native_modules!

  # Flags change depending on the env values.
  flags = get_default_flags()

  use_react_native!(
    :path => config[:reactNativePath],
    # Hermes is now enabled by default. Disable by setting this flag to false.
    # Upcoming versions of React Native may rely on get_default_flags(), but
    # we make it explicit here to aid in the React Native upgrade process.
    :hermes_enabled => flags[:hermes_enabled],
    :fabric_enabled => flags[:fabric_enabled],
    # Enables Flipper.
    #
    # Note that if you have use_frameworks! enabled, Flipper will not work and
    # you should disable the next line.
    :flipper_configuration => flipper_config,
    # An absolute path to your application root.
    :app_path => "#{Pod::Config.instance.installation_root}/.."
  )
```

Reply:
Great! :D thank you for reviewing and double-checking this!